src/commands/downloadDevContainer.ts

```diff
@@ -0,0 +1,63 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { WorkspaceFolder } from "vscode";
+import { IActionContext } from "vscode-azureextensionui";
+import { getDevContainerName } from '../downloadAzureProject/setupProjectFolder';
+import { ext } from '../extensionVariables';
+import { localize } from "../localize";
+import { ProductionSlotTreeItem } from '../tree/ProductionSlotTreeItem';
+import { SlotTreeItemBase } from '../tree/SlotTreeItemBase';
+import { requestUtils } from '../utils/requestUtils';
+import { selectWorkspaceFolder } from '../utils/workspace';
+import { tryGetFunctionProjectRoot } from './createNewProject/verifyIsProject';
+
+
+export async function downloadDevContainer(context: IActionContext, node?: SlotTreeItemBase): Promise<void> {
+    if (!node) {
+        node = await ext.tree.showTreeItemPicker<SlotTreeItemBase>(ProductionSlotTreeItem.contextValue, context);
+    }
+
+    const language: string = await node.getApplicationLanguage();
+    const devContainerName = getDevContainerName(language);
+    if (devContainerName && node.site.reserved) {
+        const message: string = localize('selectDevContainer', 'Select the destination file for the dev container folder.');
+        let devContainerFolderPath: string = await getDevContainerFolder(context, message);
```

Comment:
right below this, add something like.. 

`devContainerFolderPath = !devContainerFolderPath ? '' : devContainerFolderPath`

this will avoid any weird nullref issue, incase if the result of the await call is null or undefined.